github/data_source_github_app_token.go

```diff
@@ -0,0 +1,67 @@
+package github
+
+import (
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceGithubAppToken() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGithubAppTokenRead,
+
+		Schema: map[string]*schema.Schema{
+			"app_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions["app_auth.id"],
+			},
+			"installation_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions["app_auth.installation_id"],
+			},
+			"pem_file": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions["app_auth.pem_file"],
+			},
+			"base_url": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "https://api.github.com/",
+				Description: descriptions["base_url"],
+			},
+			"token": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func dataSourceGithubAppTokenRead(d *schema.ResourceData, meta interface{}) error {
```

Comment:
Can you create appropriate testing for this please?

Reply:
To test this I need a proper GitHub App. Would you say its a good idea to use the same setup that already exists to test the providers app auth?

https://github.com/integrations/terraform-provider-github/blob/main/github/apps_test.go

Reply:
I'm :+1: on reusing the same testing configuration.

Reply:
I found some issues with how to test this. Either we do this with real credentials which is not present in the current test scope. Or we solve this like the provider tests and mock the client response. The issue is that using httptest would require changing the base url, which is set in the provider scope. 

Do you have any other good idea for testing this datasource?

Reply:
@kfcampbell any thoughts?

Reply:
I like the idea of mocking the client response. 

> The issue is that using httptest would require changing the base url, which is set in the provider scope.

I'm not quite sure changing the base URL is an issue in tests. We have several tests that change the base URL, like [this one](https://github.com/integrations/terraform-provider-github/blob/f280fb53922abeb72cbd3ccc29beea53374b8522/github/data_source_github_repository_file_test.go#L277).

Reply:
@kfcampbell thanks for pointing me in the right direction. I have now implemented a test that validates the token creation with the mocked API endpoint. I removed any JWT validation because it would be impossible to actually have a static JWT. Hopefully this test is good enough.